core/server/api/v2/utils/serializers/output/all.js

```diff
@@ -0,0 +1,25 @@
+const debug = require('ghost-ignition').debug('api:v2:utils:serializers:output:all');
+const _ = require('lodash');
+
+const removeXBY = (object) => {
+    _.each(object, (value, key) => {
+        // CASE: go deeper
+        if (_.isObject(object[key]) || _.isArray(object[key])) {
+            removeXBY(object[key]);
+        } else if (['updated_by', 'created_by', 'published_by'].includes(key)) {
+            delete object[key];
+        }
+    });
+
+    return object;
+};
+
+module.exports = {
+    after(apiConfig, frame) {
+        debug('all after');
+
+        if (frame.response) {
+            frame.response = removeXBY(frame.response);
```

Comment:
Yeah. It's a catch 22 with v0.1 support. 

This logic is very similar in a sense that we don't want those fields exposed to the outside world, but in one case we filter it on model level (user's password filed) and for other (general x_by fields) we do it on a higher level - serializers.

Maybe later we could experiment with separate public models with api specific 'toJSON' methods. Don't have a good recipe for this yet :)

Reply:
IMO `toJSON` is model specific, serializers are API specific. `toJSON` is used for all API versions. 

> we filter it on model level (user's password filed)

IMO this is a general security feature and belongs right now to the model.

But it might be that we will move more and more stuff from `toJSON` to the API level e.g. shared serializers for all api versions. Will see what makes sense 🙂